Assets/Scripts/ROS Scripts/ROS Connections v2/Sensors/CameraSensor_ROSSensorConnection.cs

```diff
@@ -0,0 +1,155 @@
+
+        foreach (string subscriber in sensorSubscribers)
+        {
+            string subscriberTopic = "";
+
+            switch (subscriber)
+            {
+                default:
+                    subscriberTopic = "/dji_sdk/" + subscriber;
+                    ImageVisualizer imageVisualizer = new ImageVisualizer();
+                    //imageVisualizer.SetParentTransform(this.transform);
+                    imageVisualizers.Add(subscriberTopic, imageVisualizer);
+                    Debug.Log("Camera subscribing to: " + subscriberTopic);
+                    ros.AddSubscriber("/dji_sdk/" + subscriber, this);
+                    break;
+            }
+            Debug.Log("Camera Subscribing to : " + subscriberTopic);
+            sensorSubscriberTopics.Add(subscriberTopic);
+        }
+
+        ros.Connect();
+
+    }
+
+    // Update is called once per frame in Unity
+    void Update()
+    {
+        if (ros != null)
+        {
+            ros.Render();
+        }
+    }
+
+    public string GetSensorName()
+    {
+        return this.gameObject.name;
+    }
+
+    public string GetVideoType()
+    {
+        return videoType;
+    }
+
+    public void SetVideoType(string input)
+    {
+        videoType = input;
+    }
+    /// <summary>
+    /// Returns a list of connected subscriber topics (which are unique identifiers).
+    /// </summary>
+    /// <returns></returns>
+    public List<string> GetSensorSubscribers()
+    {
+        return sensorSubscriberTopics;
+    }
+
+    /// <summary>
+    /// Function to disconnect a specific subscriber
+    /// </summary>
+    /// <param name="subscriberID"></param>
+    public void Unsubscribe(string subscriberTopic)
+    {
+        if (sensorSubscriberTopics.Contains(subscriberTopic))
+        {
+            sensorSubscriberTopics.Remove(subscriberTopic);
+            ros.RemoveSubscriber(subscriberTopic);
+        }
+        else
+        {
+            Debug.Log("No such subscriber exists: " + subscriberTopic);
+        }
+
+    }
+
+    /// <summary>
+    /// Function to connect a specific subscriber
+    /// </summary>
+    /// <param name="subscriberID"></param>
+    public void Subscribe(string subscriberTopic)
+    {
+        if (sensorSubscriberTopics.Contains(subscriberTopic) == false)
+        {
+            ros.AddSubscriber(subscriberTopic, this);
+        }
+        else
+        {
+            Debug.Log("Subscriber already registered: " + subscriberTopic);
+        }
+    }
+
+    // ROS Topic Subscriber methods
+    public ROSBridgeMsg OnReceiveMessage(string topic, JSONNode raw_msg, ROSBridgeMsg parsed = null)
+    {
+        Debug.Log("Camera Recieved message");
+
+        ROSBridgeMsg result = null;
+      
+        ImageMsg meshMsg = new ImageMsg(raw_msg);
+        // Obtain visualizer for this topic
+        ImageVisualizer visualizer = imageVisualizers[topic];
+        this.imageVisualizers[topic].SetFrame(meshMsg, videoType);
```

Comment:
Seems good for now but in the future, we might need to change videoType its a bit fixed right now, no guarantees that the enum will be valid in the future.